### PR TITLE
#2832 - Statistics - Daily quarantine statistics for Cases and Contacts

### PR DESCRIPTION
**Pull request by sergiupacurariu**

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java [hunk 1]
```diff
@@ -872,6 +873,73 @@ public long count(CaseCriteria caseCriteria) {
 		return result;
 	}
 
+	public List<DashboardQuarantineDataDto> getCaseInQuarantineForDashBoard(
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java:876]: Please rename to `getQuarantineDataForDashboard` or something similar as you don't return case

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java [hunk 2]
```diff
@@ -578,6 +523,82 @@ private Predicate buildRelevantContactsFilter(Person person, Disease disease, Da
...
 	}
 
+	public Predicate getRegionDistrictDiseasePredicate(
+		Region region,
+		District district,
+		Disease disease,
+		CriteriaBuilder cb,
+		Root<Contact> contact,
+		Join<Contact, Case> caze,
+		Predicate filter) {
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java:533]: please refactor this to not use the `filter` argument

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java [hunk 3]
```diff
@@ -1009,6 +1012,24 @@ public boolean isDeleted(String contactUuid) {
 		return contactService.getContactsForDashboard(region, district, disease, from, to, user);
 	}
 
+	@Override
+	public List<DashboardQuarantineDataDto> getContactsInQuarantineForDashBoard(
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java:1016]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java [hunk 4]
```diff
@@ -578,6 +523,82 @@ private Predicate buildRelevantContactsFilter(Person person, Disease disease, Da
...
+			if (filter != null) {
+				filter = cb.and(filter, diseaseFilter);
+			} else {
+				filter = diseaseFilter;
+			}
+		}
+		return filter;
+	}
+
+	public List<DashboardQuarantineDataDto> getContactsInQuarantineForDashBoard(
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java:567]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java [hunk 5]
```diff
@@ -578,6 +523,82 @@ private Predicate buildRelevantContactsFilter(Person person, Disease disease, Da
...
+		Disease disease,
+		CriteriaBuilder cb,
+		Root<Contact> contact,
+		Join<Contact, Case> caze,
+		Predicate filter) {
+		if (region != null) {
+			Predicate regionFilter = cb.or(
+				cb.equal(contact.get(Contact.REGION), region),
+				cb.and(cb.isNull(contact.get(Contact.REGION)), cb.equal(caze.get(Case.REGION), region)));
+			if (filter != null) {
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java:538]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java [hunk 6]
```diff
@@ -578,6 +523,82 @@ private Predicate buildRelevantContactsFilter(Person person, Disease disease, Da
...
+			} else {
+				filter = regionFilter;
+			}
+		}
+
+		if (district != null) {
+			Predicate districtFilter = cb.or(
+				cb.equal(contact.get(Contact.DISTRICT), district),
+				cb.and(cb.isNull(contact.get(Contact.DISTRICT)), cb.equal(caze.get(Case.DISTRICT), district)));
+			if (filter != null) {
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java:549]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java [hunk 7]
```diff
@@ -578,6 +523,82 @@ private Predicate buildRelevantContactsFilter(Person person, Disease disease, Da
...
+				filter = cb.and(filter, districtFilter);
+			} else {
+				filter = districtFilter;
+			}
+		}
+
+		if (disease != null) {
+			Predicate diseaseFilter = cb.equal(contact.get(Contact.DISEASE), disease);
+			if (filter != null) {
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java:558]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java:876]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java:533]: changed

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**